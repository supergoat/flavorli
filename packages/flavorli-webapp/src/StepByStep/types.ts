import {IColor} from '@flavorli/elements/lib/theme/colors';

export interface ITag {
  text: string;
  color: IColor;
}

export interface ILink {
  heading: string;
  name: string;
  from: number;
  timerId?: number;
}

export interface IIngredient {
  qty: string;
  name: string;
}

export interface ITimer {
  id: number;
  name: string;
  minutes: number;
  seconds: number;
  isPaused?: boolean;
}

export interface IKitchenware {
  qty: string;
  name: string;
}

export interface IImage {
  src: string;
  alt: string;
}
export interface IStep {
  type: 'PREPARATION' | 'MISE_EN_PLACE_STEP';
  no: number;
  description: string;
  tag: ITag;
  links: ILink[];
  ingredients: IIngredient[];
  timer?: ITimer;
  kitchenware: IKitchenware[];
  images?: IImage[];
}

export interface IItem {
  qty: string;
  name: string;
}

export interface IMiseEnPlaceStep {
  no: number;
  type: 'MISE_EN_PLACE';
  heading: string;
  image: string;
  items: IItem[];
}

export interface IIntro {
  type: 'INTRO';
  name: string;
  image: string;
  preparation: string;
  cooking: string;
  portions: string;
  difficulty: string;
}
