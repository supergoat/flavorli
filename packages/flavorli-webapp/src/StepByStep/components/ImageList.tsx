import React from 'react';
import {Stack} from '@flavorli/elements';

/**
 * Implemented according to
 * @see https://w3c.github.io/aria-practices/#carousel
 */
export interface IImageListProps {
  children: React.ReactNode;
}
const ImageList = ({children}: IImageListProps) => {
  return (
    <Stack
      width="100%"
      height="100%"
      id="step-images"
      data-testid="step-images"
      aria-live="polite"
      paddingLeft={8}
      paddingRight={8}
    >
      {children}
    </Stack>
  );
};

export default ImageList;
