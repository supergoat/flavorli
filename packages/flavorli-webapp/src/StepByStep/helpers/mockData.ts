import {IStep} from '../types';

export const steps: IStep[] = [
  {
    no: 1,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description: 'Place a pot of water in medium heat and set aside to boil',
    ingredients: [],
    kitchenware: [
      {
        qty: '1',
        name: 'Pot',
      },
    ],
  },
  {
    no: 2,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description: 'Place a sauce pan over high heat and add the olive oil',
    ingredients: [
      {
        qty: '4 - 6 tbsps',
        name: 'olive oil',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Sauce Pan',
      },
    ],
  },
  {
    no: 3,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description:
      'Cut the eggplan into 1 cm cubes. Add it to the pan and sauté for 1-2 minutes',
    ingredients: [
      {
        qty: '2',
        name: 'eggplants',
      },
    ],
    kitchenware: [
      {qty: '1', name: 'Knife'},
      {qty: '1', name: 'Cutting Board'},
      {qty: '1', name: 'Spoon'},
    ],
  },
  {
    no: 4,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description:
      'Finely chop the onion, add it to the pan and sauté for 1-2 minutes',
    ingredients: [
      {
        qty: '1',
        name: 'onion',
      },
    ],
    kitchenware: [],
  },
  {
    no: 5,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description:
      'Cut the zucchinis and the peppers into small cubes and add them to the pan. Sauté for 4-5 minutes, until the vegetables are ready',
    ingredients: [
      {
        qty: '2',
        name: 'zucchinis',
      },
      {
        qty: '1',
        name: 'red bell pepper',
      },
      {
        qty: '1',
        name: 'green bell pepper',
      },
    ],
    kitchenware: [],
  },
  {
    no: 6,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description:
      'Add the salt, pepper, oregano, thyme, chili flakes, sugar and tomato paste and sauté',
    ingredients: [
      {
        qty: '',
        name: 'salt',
      },
      {
        qty: '',
        name: 'pepper',
      },
      {
        qty: '1 tsp',
        name: 'oregano',
      },
      {
        qty: '1 tbsp',
        name: 'thyme',
      },
      {
        qty: '1 tsp',
        name: 'chilli flakes',
      },
      {
        qty: '1 tbsp',
        name: 'granulated sugar',
      },
      {
        qty: '1 tsp',
        name: 'tomato paste',
      },
    ],
    kitchenware: [
      {
        qty: '1',
        name: 'Tea Spoon',
      },
    ],
  },
  {
    no: 7,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [],
    description:
      'Add the wine and canned tomatoes and simmer at medium-low heat for 10 minutes',
    ingredients: [
      {
        qty: '40 g',
        name: 'white wine',
      },
      {
        qty: '400 g',
        name: 'canned tomatoes',
      },
    ],
    timer: {id: 7, name: 'Step 7: Simmer Sauce', minutes: 10, seconds: 0},
    kitchenware: [{qty: '1', name: 'scale'}],
  },
  {
    no: 8,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [{from: 1, heading: 'From Step 1', name: 'Boiling water'}],
    description:
      'In the pot with boiling water, add salt along with the macaroni, and boil for ten minutes',
    ingredients: [
      {
        qty: '',
        name: 'salt',
      },
      {
        qty: '500 g',
        name: 'macaroni',
      },
    ],
    timer: {id: 8, name: 'Step 8: Cook Pasta', minutes: 10, seconds: 0},
    kitchenware: [],
  },
  {
    no: 9,
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    links: [],
    description: 'Preheat the oven at 180 C set to fan',
    ingredients: [],
    kitchenware: [],
  },
  {
    no: 10,
    tag: {
      text: 'Bechamel',
      color: 'tagOrange',
    },
    links: [],
    description:
      'Place a pot over medium heat. Add the olive oil and the flour and using a hand whisk, mix well until all of the flour is absorbed',
    ingredients: [
      {
        qty: '80 g',
        name: 'olive oil',
      },
      {
        qty: '80 g',
        name: 'all-purpose flour',
      },
    ],
    kitchenware: [
      {qty: '1', name: 'Small Pot'},
      {qty: '1', name: 'Hand Whisk'},
    ],
  },
  {
    no: 11,
    tag: {
      text: 'Bechamel',
      color: 'tagOrange',
    },
    links: [],
    description: 'Add the soy milk in 7-8 batches, stirring constantly',
    ingredients: [
      {
        qty: '800 g',
        name: 'soy milk',
      },
    ],
    kitchenware: [],
  },
  {
    no: 12,
    tag: {
      text: 'Bechamel',
      color: 'tagOrange',
    },
    links: [],
    description:
      'Season with salt and pepper and as soon as it thickens and comes to boil, remove from the heat',
    ingredients: [
      {
        qty: '',
        name: 'salt',
      },
      {
        qty: '',
        name: 'pepper',
      },
    ],
    kitchenware: [],
  },
  {
    no: 13,
    tag: {
      text: 'Sauce',
      color: 'tagRed',
    },
    links: [
      {
        from: 7,
        heading: 'From Step 7',
        name: 'Pan with vegetables',
        timerId: 7,
      },
      {
        from: 8,
        heading: 'From Step 8',
        name: 'Macaroni',
        timerId: 8,
      },
    ],
    description:
      'Drain the macaroni and transfer to the pan with the vegetables. Mix and set aside',
    ingredients: [],
    kitchenware: [{qty: '1', name: 'Colander'}],
  },
  {
    no: 14,
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    links: [{from: 12, heading: 'From Step 12', name: 'Bechamel sauce'}],
    description:
      'Add 100g of the bechamel into the pasta mixture, add oregano, mint, and mix well',
    ingredients: [
      {
        qty: '1 tsp',
        name: 'oregano',
      },
      {
        qty: '1 tbsp',
        name: 'mint',
      },
    ],
    kitchenware: [],
  },
  {
    no: 15,
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    links: [],
    description:
      'Grease 40 cm ovenproof baking dish with olive oil and sprinke the breadcrumps',
    ingredients: [
      {
        qty: '1 tbsp',
        name: 'olive oil',
      },
      {
        qty: '25 g',
        name: 'breadcrumps',
      },
    ],
    kitchenware: [{qty: '1', name: '40 cm ovenproof baking dish'}],
  },
  {
    no: 16,
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    links: [],
    description:
      'Add the pasta mixture, spread the bechamel sauce, and sprinke the breadcrumps, olive oil, thyme and pepper',
    ingredients: [
      {
        qty: '25 g',
        name: 'breadcrumps',
      },
      {
        qty: '1 tbsp',
        name: 'olive oil',
      },
      {
        qty: '1 tbsp',
        name: 'thyme',
      },
      {
        qty: '',
        name: 'pepper',
      },
    ],
    kitchenware: [],
  },
  {
    no: 17,
    tag: {
      text: 'Assembling',
      color: 'tagPink',
    },
    links: [],
    description: 'Bake for 25 - 30 minutes. Remove and let it cool',
    ingredients: [],
    timer: {id: 17, name: 'Step 17: Bake Pastitsio', minutes: 30, seconds: 0},
    kitchenware: [
      {
        qty: '1',
        name: 'Oven Mitts',
      },
    ],
  },
];
