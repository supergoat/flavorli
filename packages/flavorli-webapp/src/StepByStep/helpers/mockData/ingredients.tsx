export const ingredients = [
  {
    qty: '2',
    name: 'eggplants',
  },
  {
    qty: '1',
    name: 'onion',
  },
  {
    qty: '1 clove',
    name: 'garlic',
  },
  {
    qty: '2',
    name: 'zucchinis',
  },
  {
    qty: '1',
    name: 'red bell pepper',
  },
  {
    qty: '1',
    name: 'green bell pepper',
  },
  {
    qty: '2 teaspoon',
    name: 'oregano',
  },
  {
    qty: '2 tablespoon',
    name: 'thyme',
  },
  {
    qty: '1 teaspoon',
    name: 'chilli flakes',
  },
  {
    qty: '1 tablespoon',
    name: 'granulated sugar',
  },
  {
    qty: '1 tablespoon',
    name: 'tomato paste',
  },
  {
    qty: '50 g',
    name: 'white wine',
  },
  {
    qty: '400 g',
    name: 'canned tomatoes',
  },
  {
    qty: '500 g',
    name: 'macaroni',
  },
  {
    qty: '80 g',
    name: 'all-purpose flour',
  },
  {
    qty: '80 g',
    name: 'olive oil',
  },
  {
    qty: '800 g',
    name: 'soy milk',
  },
  {
    qty: '50 g',
    name: 'dry breadcrumps',
  },
  {
    qty: '1 tablespoon',
    name: 'mint',
  },
  {
    qty: '1',
    name: 'squeeze bottle olive oil',
  },
  {
    qty: '',
    name: 'salt',
  },
  {
    qty: '',
    name: 'pepper',
  },
];
