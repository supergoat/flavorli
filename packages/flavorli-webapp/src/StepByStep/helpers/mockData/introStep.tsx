import {IIntro} from '../../types';

export const introStep: IIntro = {
  name: 'Greek Pastitsio',
  image:
    'https://akispetretzikis.com/system/uploads/medium/data/14955/recipe_main_vegan-pastitsio.jpg',
  preparation: '25 minutes',
  cooking: '40 minutes',
  portions: '8 - 10',
  difficulty: 'Level 2',
};
