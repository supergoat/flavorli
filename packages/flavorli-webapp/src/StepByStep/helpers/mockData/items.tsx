export const items = [
  {
    qty: '2',
    name: 'Bowls',
  },
  {
    qty: '1',
    name: 'Small Bowl',
  },
  {
    qty: '3',
    name: 'Glasses',
  },
  {
    qty: '11',
    name: 'Ramekins',
  },
  {
    qty: '1',
    name: 'Sauce Pan',
  },
  {
    qty: '1',
    name: 'Pot',
  },
  {
    qty: '1',
    name: 'Small Pot',
  },
  {
    qty: '1',
    name: 'Cutting Board',
  },
  {
    qty: '1',
    name: 'Scale',
  },
  {
    qty: '1',
    name: 'Hand Whisk',
  },
  {
    qty: '1',
    name: 'Colander',
  },
  {
    qty: '1',
    name: '40 cm ovenproof dish',
  },
];
