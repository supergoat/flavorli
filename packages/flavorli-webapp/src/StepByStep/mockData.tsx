import {IStep, IRecipeTimer, ILink, IRecipe} from '../types';

export const recipes: IRecipe[] = [
  {
    id: '1',
    createdAt: '2020-01-19T19:37:42.873Z',
    author: 'Robin Robertson',
    name: 'Cherry Berry Smoothie',
    image:
      'https://flavorli-bucket-2.s3.eu-west-2.amazonaws.com/CherryBerrySmoothie.jpeg',
    preparationTime: 7,
    cookingTime: 0,
    portions: '1',
    difficulty: '1',
    ingredients: [
      {
        qty: '1 cup/100 g',
        name: 'Frozen Blueberries',
      },
      {
        qty: '1/2 cup/110 g',
        name: 'fresh or frozen pitted cherries',
      },
      {
        qty: '1',
        name: 'frozen ripe banana, cut into chunks before freezing',
      },
      {
        qty: '1 tbsp',
        name: 'ground flaxseeds (or linseeds)',
      },
      {
        qty: '1 tbsp',
        name: 'almond butter (or peanut butter)',
      },
      {
        qty: '1 cup/22.5 g',
        name: 'raw spinach leaves',
      },
      {
        qty: '1 1/2 cup (360 ml)',
        name: 'water',
      },
    ],
    items: [
      {
        qty: '1',
        name: 'Blender',
      },
      {
        qty: '1',
        name: 'Cup',
      },
      {
        qty: '1/2',
        name: 'cup',
      },
      {
        qty: '1',
        name: 'tablespoon',
      },
    ],
    steps: [
      {
        no: 1,
        type: 'MISE_EN_PLACE',
        tag: {
          text: 'Smoothie',
          color: 'tagRed',
        },
        items: [
          {
            qty: '1',
            name: 'Blender',
          },
        ],
        ingredients: [
          {
            qty: '1 cup/100 g',
            name: 'Frozen Blueberries',
          },
          {
            qty: '1/2 cup/110 g',
            name: 'fresh or frozen pitted cherries',
          },
          {
            qty: '1',
            name: 'frozen ripe banana, cut into chunks before freezing',
          },
          {
            qty: '1 tbsp',
            name: 'ground flaxseeds (or linseeds)',
          },
          {
            qty: '1 tbsp',
            name: 'almond butter (or peanut butter)',
          },
          {
            qty: '1 cup/22.5 g',
            name: 'raw spinach leaves',
          },
          {
            qty: '1 1/2 cup (360 ml)',
            name: 'water',
          },
        ],
        tasks: [
          'Add 1 cup/100 g Frozen Blueberries in a blender',
          'Add 1/2 cup/110 g fresh or frozen pitted cherries',
          'Add 1 frozen ripe banana, cut into chunks before freezing',
          'Add 1 tbsp ground flaxseeds (or linseeds)',
          'Add 1 tbsp almond butter (or peanut butter)',
          'Add 1 cup/22.5 g raw spinach leaves',
          'Add 1 1/2 cup cup (360 ml) water',
        ],
        images: [],
      },
      {
        no: 2,
        type: 'PREPARATION',
        tag: {
          text: 'Smoothie',
          color: 'tagRed',
        },
        items: [],
        ingredients: [],
        tasks: [
          'Blend all the ingredients until smooth and creamy for about 1 minute',
          'Add more water if you prefer a thinner consistency',
          'Serve immediately',
        ],
        timer: {
          id: '2',
          name: 'Step 2: Blend',
          minutes: 1,
          seconds: 0,
        },
        images: [],
      },
    ],
  },
];

export const timer: IRecipeTimer = {
  id: '1',
  name: 'Thicken Sauce',
  minutes: 10,
  seconds: 0,
  isPaused: true,
  updatedAt: '2020-01-26T15:15:06.689Z',
  remainingTime: 600000,
};

export const link: ILink = {
  from: 1,
  heading: 'From Step 1',
  name: 'Thicken Sauce',
};

export const linkWithTimer: ILink = {
  from: 1,
  heading: 'From Step 1',
  name: 'Thicken Sauce',
  timerId: '1',
};

export const items = [
  {
    qty: '1',
    name: 'Pot',
  },
];

export const preparationStep: IStep = {
  type: 'MISE_EN_PLACE',
  tag: {
    text: 'Sauce',
    color: 'tagRed',
  },
  tasks: ['Add the coconul oil in a ramekin and place it first in order'],
  ingredients: [
    {
      qty: '2 tbsps',
      name: 'coconut oil',
    },
  ],
  items: [
    {
      qty: '1',
      name: 'Ramekin',
    },
  ],
  images: [],
};

export const recipeStep: IStep = {
  type: 'PREPARATION',
  no: 1,
  tag: {
    text: 'Sauce',
    color: 'tagRed',
  },
  links: [],
  tasks: ['Heat the coconut oil in a small pot on low heat'],
  ingredients: [
    {
      qty: '2 tbsps',
      name: 'coconut oil',
    },
  ],
  items: [
    {
      qty: '1',
      name: 'Small Pot',
    },
  ],
  images: [],
};

export const recipeSteps = [recipeStep];

export const introStep = {
  author: 'Akis Petratzikis',
  name: 'Greek Pastitsio',
  image: '',
  preparation: '25 minutes',
  cooking: '40 minutes',
  portions: '8 - 10',
  difficulty: 'Level 2',
};

export const ingredients = [{qty: '1', name: 'Onion'}];

export const images = [
  {
    src: '',
    alt: 'image 1',
  },

  {
    src: '',
    alt: 'image 2',
  },
];
