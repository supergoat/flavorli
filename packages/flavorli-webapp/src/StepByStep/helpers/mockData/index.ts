// https://www.youtube.com/watch?v=q1q3HfVvzQ0
import {ingredients} from './ingredients';
import {items} from './items';
import {preparationSteps} from './preparationSteps';
import {recipeSteps} from './recipeSteps';
import {IStepByStep} from '../../types';

export const stepByStep: IStepByStep = {
  intro: {
    name: 'Greek Pastitsio',
    image:
      'https://akispetretzikis.com/system/uploads/medium/data/14955/recipe_main_vegan-pastitsio.jpg',
    preparation: '25 minutes',
    cooking: '40 minutes',
    portions: '8 - 10',
    difficulty: 'Level 2',
  },
  ingredients,
  items,
  preparationSteps,
  recipeSteps,
};

export {ingredients, items, preparationSteps, recipeSteps};
