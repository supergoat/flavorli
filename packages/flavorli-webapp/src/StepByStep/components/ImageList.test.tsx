import React from 'react';
import {axe} from 'jest-axe';
import userEvent from '@testing-library/user-event';
import {render} from '../../helpers/test-helpers';
import ImageList from './ImageList';
import {IImage} from '../types';

import {images} from '../mockData';

const setup = (customImages?: IImage[]) => {
  return {
    ...render(
      <>
        {/* Add a div with id step-images to be used by aria-controls */}
        <div id="step-image" />
        <ImageList images={customImages || images} />
      </>,
    ),
    images,
  };
};

describe('Step', () => {
  it('should not have any axe violations', async () => {
    const {container} = setup();
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should use a section element to designate ImageList as land mark region, with an aria-lable to describe its content', () => {
    const {getByLabelText} = setup();
    const section = getByLabelText('List of step images');

    expect(section.tagName).toEqual('SECTION');
  });

  it('should render correctly', () => {
    const {container} = setup();
    expect(container).toMatchSnapshot();
  });

  it('should render an empty div if images is an empty array', () => {
    const {container} = setup([]);
    expect(container).toMatchSnapshot();
  });

  it('should display only one image at a time', async () => {
    const {queryByAltText, images} = setup();
    const [image1, ...rest] = images;
    expect(queryByAltText(`${image1.alt}`)).not.toBeNull();

    rest.forEach(image => expect(queryByAltText(image.alt)).toBeNull());
  });

  it('should have a next button that when clicked hides the current image and brings the next image into view', () => {
    const {queryByAltText, getByLabelText, images} = setup();

    const image1 = images[0];
    const image2 = images[1];

    expect(queryByAltText(`${image1.alt}`)).not.toBeNull();

    const nextButton = getByLabelText(/Next Image/i);

    userEvent.click(nextButton);

    expect(queryByAltText(`${image2.alt}`)).not.toBeNull();

    expect(queryByAltText(`${image1.alt}`)).toBeNull();
  });

  it('should have a previous button that when clicked hides the current image and brings the previous image into view', () => {
    const {queryByAltText, getByLabelText, images} = setup();

    const image1 = images[0];
    const image2 = images[1];

    const nextButton = getByLabelText(/Next Image/i);

    userEvent.click(nextButton);

    expect(queryByAltText(`${image2.alt}`)).not.toBeNull();

    const previousButton = getByLabelText(/Previous Image/i);

    userEvent.click(previousButton);

    expect(queryByAltText(`${image1.alt}`)).not.toBeNull();

    expect(queryByAltText(`${image2.alt}`)).toBeNull();
  });

  it('should have a role of group on the parent of the image enable assistive technology users to preceive the boundaries of the image', () => {
    const {getByRole} = setup();
    getByRole('group');
  });

  it('should have an aria-label to indicate to assistive technology users the image they are on', () => {
    const {getByLabelText, images} = setup();
    getByLabelText(`Image 1 of ${images.length}`);
  });

  it('should have a next button with an accessible label', () => {
    const {getByLabelText} = setup();

    getByLabelText(/Next Image/i);
  });

  it('should have a next button with aria-controls set to "step-images" to inform assistive technology users that it controls which image is displayed', () => {
    const {getByLabelText} = setup();

    const nextStepButton = getByLabelText(/Next Image/i);

    expect(nextStepButton).toHaveAttribute('aria-controls', 'step-images');
  });

  it('it should hide the next button when the current step is the last step', () => {
    const image = {src: 'fake-src', alt: ''};
    const {queryByLabelText} = setup([image]);
    expect(queryByLabelText(/Next Image/i)).not.toBeVisible();
  });

  it('should have a previous button with an accessible label', () => {
    const {getByLabelText} = setup();

    getByLabelText(/Previous Image/i);
  });

  it('should have a previous button with aria-controls set to "step-images" to inform assistive technology users that it controls which image is displayed', () => {
    const {getByLabelText} = setup();

    const previousStepButton = getByLabelText(/Previous Image/i);

    expect(previousStepButton).toHaveAttribute('aria-controls', 'step-images');
  });

  it('it should hide the previous button when the current step is the first step', () => {
    const image = {src: 'fake-src', alt: ''};
    const {queryByLabelText} = setup([image]);
    expect(queryByLabelText(/Previous Image/i)).not.toBeVisible();
  });
});
