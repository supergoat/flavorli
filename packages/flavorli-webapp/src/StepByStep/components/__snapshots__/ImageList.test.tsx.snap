// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step should render an empty div if images is an empty array 1`] = `
<div>
  <div
    id="step-image"
  />
</div>
`;

exports[`Step should render correctly 1`] = `
<div>
  <div
    id="step-image"
  />
  <section
    aria-label="List of step images"
    class="sc-fzXfLO VjJSW"
  >
    <div
      aria-live="polite"
      class="sc-AykKH kSMEZQ"
      data-testid="step-images"
      direction="vertical"
      height="300px"
      id="step-images"
      width="300px"
    >
      <div
        aria-label="Image 1 of 2"
        class="sc-AykKH csfDLG"
        direction="vertical"
        height="100%"
        role="group"
        width="100%"
      >
        <img
          alt="image 1"
          class="sc-fzXfLP iPnlFu"
          src=""
        />
      </div>
      <button
        aria-controls="step-images"
        aria-label="Previous Image"
        class="sc-fzXfLQ sc-fzXfLR hrlKtg"
      >
        <img
          alt=""
          src="right_chevron_primary.svg"
        />
      </button>
      <button
        aria-controls="step-images"
        aria-label="Next Image"
        class="sc-fzXfLQ sc-fzXfLS GOLfT"
      >
        <img
          alt=""
          src="right_chevron_primary.svg"
        />
      </button>
    </div>
  </section>
</div>
`;
